hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyRequest.java

```diff
@@ -1051,4 +1057,11 @@ protected void filterOutBlocksStillInUse(OmKeyInfo referenceKey,
     LOG.debug("After block filtering,  keysToBeFiltered = {}",
         keysToBeFiltered);
   }
+
+  protected void validateEncryptionKeyInfo(OmBucketInfo bucketInfo, KeyArgs keyArgs) throws OMException {
+    if (bucketInfo.getEncryptionKeyInfo() != null && !keyArgs.hasFileEncryptionInfo()) {
+      throw new OMException("Attempting to create unencrypted file " +
+          keyArgs.getKeyName() + " in encrypted bucket " + keyArgs.getBucketName(), INTERNAL_ERROR);
```

Comment:
Can we use a different error code here (maybe a new one?) While the cause of this issue would a bug inside the OM, INTERNAL_ERROR and METADATA_ERROR cause the OM to crash. So if we hit a bug in this code path, the whole cluster will be inoperable until patched. Probably better to fail the request so we can identify the error, but leave the OMs running.